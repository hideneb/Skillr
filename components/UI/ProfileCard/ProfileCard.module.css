@value md-up from "../../../styles/Breakpoints.module.css";

.profileCardContainer {
    display: flex;
    flex-wrap: wrap;
    padding: 12px;
    width: 100%;
    margin-bottom: 50px;
    border-radius: 12px;
    box-shadow: 0px 4px 16px rgba(0, 0, 0, 0.12);
}

.profileDetailsCardImgContainer {
    height: 56px;
    width: 56px;
    min-width: 56px;
    overflow: hidden;
    margin-right: 20px;
}

.profileDetailsCardImgContainer img {
    object-fit: cover;
    border-radius: 5px;
}

.profileCardTextContainer {
    flex-grow: 1;
}

.title {
    font-family: 'RedHatDisplay';
    font-size: 20px;
    font-weight: 700;
    color: var(--rich-blue-8);
}

.videosTitle {
    font-family: 'RedHatDisplay';
    font-size: 20px;
    font-weight: 700;
    color: var(--rich-blue-8);
    margin-bottom: 12px;
}

.rateContainer {
    display: flex;
}

.rate {
    display: flex;
    align-items: center;
    margin-right: 24px;
    color: var(--rich-blue-8);
    font-family: 'Poppins';
    font-size: 12px;
    font-weight: 500;
}

.profileCardDescription {
    display: flex;
    margin-bottom: 26px;
    font-family: 'Poppins';
    font-size: 16px;
    line-height: 24px;
    color: var(--rich-blue-7);
}

.saveButton {
    background: var(--skillr-pink-1);
    height: 40px;
    width: 100px;
    color: white;
    border-radius: 8px;
}

.cancelButton {
    border: 1px solid lightgray;
    height: 40px;
    color: gray;
    width: 100px;
    border-radius: 8px;
}

.videos {
    margin-top: 16px;
}

@media md-up {
    .profileCardContainer {
        flex-wrap: nowrap;
        padding: 26px 32px;
        box-shadow: 0px 4px 16px rgba(0, 0, 0, 0.12);
    }

    .profileDetailsCardImgContainer {
        width: 76px;
        height: 76px;
    }

    .title {
        font-size: 21px;
    }

    .rate {
        font-size: 14px;
    }
}
