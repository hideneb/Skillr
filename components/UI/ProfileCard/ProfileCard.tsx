import React from 'react';
import styles from './ProfileCard.module.css';
import cx from 'classnames';
import { SkillrDto } from '@/lib/types/skillr';
import ProfileVideos from '../ProfileVideos';

type ProfileCardProps = {
    imgSrc: string;
    name: string;
    description: string;
    ratePerMinute: number;
    skillr: SkillrDto;
    featureImage: string;
};

const ProfileCard: React.FC<ProfileCardProps> = ({
    imgSrc,
    name,
    description,
    ratePerMinute,
    skillr,
    featureImage,
}) => {
    const videos = skillr.images.filter((i) => !!i.video);

    return (
        <div className={cx(styles.card, styles.profileCardContainer)}>
            <div className="d-flex flex-column" style={{ flex: 1, maxWidth: '100%' }}>
                <p className={styles.profileCardDescription}>{description}</p>
                <div className="d-flex">
                    <div className={styles.profileDetialsCardImgContainer}>
                        {imgSrc ? <img height={'100%'} width={'100%'} alt="" src={imgSrc} /> : null}
                    </div>
                    <div className="d-flex" style={{ flexGrow: 1 }}>
                        <div className={styles.profileCardTextContainer}>
                            <div className={styles.title}>{name}</div>
                            <div className={styles.rateContainer}>
                                {ratePerMinute ? (
                                    <div className={styles.rate}>${(ratePerMinute / 100).toFixed(2)}/min</div>
                                ) : (
                                    ''
                                )}
                            </div>
                        </div>
                    </div>
                </div>
                {videos.length > 0 && (
                    <div className={styles.videos}>
                        <div className={styles.videosTitle}>Videos</div>
                        <ProfileVideos featureImage={featureImage} videos={videos} />
                    </div>
                )}
            </div>
        </div>
    );
};

export default ProfileCard;
