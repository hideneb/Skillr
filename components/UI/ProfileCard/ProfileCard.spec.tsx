import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import "@testing-library/jest-dom";
import ProfileCard from "../ProfileCard";

test("displaying a profile card", async () => {
  // Arrange

  const { container } = render(
    <ProfileCard
      about="one two three four five six seven eight nine ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen twenty one two three four five six seven eight nine ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen twenty one two three four five six seven eight nine ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen twenty one two three four five six seven eight nine ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen twenty one two three four five six seven eight nine ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen twenty one two three four five six seven eight nine ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen twenty"
      avatar="https://via.placeholder.com/150"
      displayName="Display Name"
      username="username"
    />
  );
  expect(screen.queryByText("...", { exact: false })).toBeInTheDocument();
  expect(container).toMatchSnapshot();

  // Act

  const user = userEvent.setup();
  user.click(screen.getByText(/read more/i));

  // Assert

  expect(await screen.findByText(/read less/i)).toBeInTheDocument();
  expect(screen.queryByText("...", { exact: false })).not.toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
