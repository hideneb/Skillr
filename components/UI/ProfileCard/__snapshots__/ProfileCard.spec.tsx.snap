// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displaying a profile card 1`] = `
<div>
  <div
    class="profileCard"
  >
    <div
      class="profileCardContent card"
    >
      <div
        class="profileCardImageContainer"
      >
        <img
          alt=""
          height="100%"
          src="https://via.placeholder.com/150"
          width="100%"
        />
      </div>
      <div
        class="mobileProfileHeader"
      >
        <div>
          @
          username
        </div>
        <div>
          Display Name
        </div>
      </div>
      <p
        class="profileCardAbout"
      >
        one two three four five six seven eight nine ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen twenty one two three four five six seven eight nine ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen twenty one two three four five six seven eight nine ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen twenty one two three four five six seven eight nine ten eleven twelve thirteen fourteen fifteen sixteen sevente...
      </p>
      <button
        class="readMoreButton"
      >
        Read more
      </button>
    </div>
  </div>
</div>
`;

exports[`displaying a profile card 2`] = `
<div>
  <div
    class="profileCard"
  >
    <div
      class="profileCardContent card"
    >
      <div
        class="profileCardImageContainer"
      >
        <img
          alt=""
          height="100%"
          src="https://via.placeholder.com/150"
          width="100%"
        />
      </div>
      <div
        class="mobileProfileHeader"
      >
        <div>
          @
          username
        </div>
        <div>
          Display Name
        </div>
      </div>
      <p
        class="profileCardAbout"
      >
        one two three four five six seven eight nine ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen twenty one two three four five six seven eight nine ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen twenty one two three four five six seven eight nine ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen twenty one two three four five six seven eight nine ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen twenty one two three four five six seven eight nine ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen twenty one two three four five six seven eight nine ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen twenty
      </p>
      <button
        class="readLessButton"
      >
        Read less
      </button>
    </div>
  </div>
</div>
`;
